### DISCUSSION TITLE
Версии расчетов в простой таблице

### DISCUSSION THREAD

**Discussion by LostAlly** [totumonline/totum-ru-issues-and-discussions#560] [Как это сделать?] [answered]
Приветствую
Как лучше сделать?
Есть простая таблица предварительных расчетов заказов, состав расчета хранится в поле данные и будет открываться во временной таблице и состав будет помещаться в поле данных. Бывает в предварительный расчет вносятся изменения или нужно несколько вариантов одного расчета.

Как то  можно сделать версии внутри одного поля дата хранить несколько массивов?
Или
На каждую версию создавать отдельную запись и как-то связывать все версии между собой?

как лучше и правильней это организовать. Количество вариантов изначально неизвестно.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Я бы сделал отдельную простую в которой хранил бы версии. Тогда из таблицы рассчетов можно выбирать что открыть и всякую дополнительную логику можно организовывать.

**Reply by LostAlly** [reply]
Я правильно понял, что это случае таблица версий будет состоять  в упрощенной форме из трех полей 
1. Связь с записью в таблице расчетов
2. Версия расчета
3. Поле типа данные с массивом одного расчета.
И  из таблицы расчетов можем попасть в список версий расчетов и выбрать нужную версию?
Спасибо

**Reply by ttm-alexeypolunin** [reply]
Наверное там будет еще кнопка открыть рассчет. Да. Из таблицы расчетов делаете динамическую кнопку, которая если нет рассчетов создает новый, если есть открывает ифрейм с рассчетами. Оттуда ужн открывается конкретный рассчет или добавляется новый. Можно в табл с версиями вытащить агрегированную информацию типа общей стоимости или коментария, что бы их человек различать мог.в таблице рассчетов можете на кнопке показывать сколтко вариантов есть.
